tinytuya/BulbDevice.py

```diff
@@ -377,19 +362,8 @@ def set_hsv(self, h, s, v, nowait=False):
             )
 
         (r, g, b) = colorsys.hsv_to_rgb(h, s, v)
-        hexvalue = BulbDevice._rgb_to_hexvalue(
-            r * 255.0, g * 255.0, b * 255.0, self.bulb_type
-        )
-
-        payload = self.generate_payload(
-            CONTROL,
-            {
-                self.DPS_INDEX_MODE[self.bulb_type]: self.DPS_MODE_COLOUR,
-                self.DPS_INDEX_COLOUR[self.bulb_type]: hexvalue,
-            },
-        )
-        data = self._send_receive(payload, getresponse=(not nowait))
-        return data
+
+        return self.set_colour( r * 255.0, g * 255.0, b * 255.0, nowait=nowait )

```

Comment:
Also, clean API change. I think there is enough API change here we need to signal breaking change with the version number we pick.

Reply:
I don't think there are any API changes?  Most of this is function de-duplication and code cleanup.  I'll double-check but I think the only user-visible changes are 1) manually reading `d.bulb_type` will read `None` until either `d.status()` is called or a command is sent  2) unknown bulbs now default to type 'B' instead of 'A' and  3) a `force` argument has been added to set_colour() and set_white()  to disable some of the new optimizations, however this feels like a dirty hack to me and I'm probably going to rework it to remove this.

Reply:
Fair. Those are minor. Thinking of old code out there that may be impacted by those two things is very edge case. I'm good with a minor update. I had a stronger impression of the change when I was originally doing the review, but feel less concerned now.